discord/ext/commands/converter.py

```diff
@@ -504,6 +505,25 @@ async def convert(self, ctx: Context, argument: str) -> discord.StoreChannel:
         return TextChannelConverter._resolve_channel(ctx, argument, ctx.guild.channels, discord.StoreChannel)
 
 
+class GuildChannelConverter(IDConverter[discord.abc.GuildChannel]):
+    """Converts to a :class:`~discord.abc.GuildChannel`.
+
+    All lookups are via the local guild. If in a DM context, then the lookup
+    is done by the global cache.
+
+    The lookup strategy is as follows (in order):
+
+    1. Lookup by ID.
+    2. Lookup by mention.
+    3. Lookup by name.
+
+    .. versionadded:: 2.0
+    """
+
+    async def convert(self, ctx: Context, argument: str) -> discord.abc.GuildChannel:
+        return TextChannelConverter._resolve_channel(ctx, argument, ctx.guild.channels, discord.abc.GuildChannel)
```

Comment:
I'd personally move `._resolve_channel` to this class and access it as such in the other classes.

Reply:
Moved the staticmethod into GuildChannelConverter and all references to it